src/main/kotlin/me/partlysanestudios/partlysaneskies/features/skills/SkillUpgradeWebhook.kt

```diff
@@ -0,0 +1,98 @@
+//
+// Written by Su386.
+// See LICENSE for copyright and license notices.
+//
+
+
+package me.partlysanestudios.partlysaneskies.features.skills
+
+import gg.essential.elementa.constraints.CenterConstraint
+import gg.essential.elementa.dsl.percent
+import me.partlysanestudios.partlysaneskies.PartlySaneSkies
+import me.partlysanestudios.partlysaneskies.config.psconfig.Toggle
+import me.partlysanestudios.partlysaneskies.config.psconfig.Toggle.Companion.asBoolean
+import me.partlysanestudios.partlysaneskies.features.discord.webhooks.EmbedData
+import me.partlysanestudios.partlysaneskies.features.discord.webhooks.EmbedField
+import me.partlysanestudios.partlysaneskies.features.discord.webhooks.Webhook
+import me.partlysanestudios.partlysaneskies.features.discord.webhooks.WebhookData
+import me.partlysanestudios.partlysaneskies.render.gui.components.PSSItemRender
+import me.partlysanestudios.partlysaneskies.utils.ChatUtils.trueUnformattedMessage
+import me.partlysanestudios.partlysaneskies.utils.ImageUtils.asHex
+import me.partlysanestudios.partlysaneskies.utils.StringUtils.romanNumeralToInt
+import me.partlysanestudios.partlysaneskies.utils.StringUtils.toRoman
+import net.minecraft.init.Items
+import net.minecraft.item.ItemStack
+import net.minecraftforge.client.event.ClientChatReceivedEvent
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
+import java.awt.Color
+
+object SkillUpgradeWebhook: Webhook() {
+    override val icon = PSSItemRender(ItemStack(Items.diamond_pickaxe), true)
+        .setX(CenterConstraint())
+        .setY(CenterConstraint())
+        .setWidth(90.percent)
+
+    override val id = "skillLevelUp"
+    override val name = "Skill Level Up"
+    override val description = "Send a webhook whenever you level up a skill"
+
+    init {
+        config.registerOption("multipleOf5", Toggle("Send only multiples of 5", "Only send multiples of 5 (Lvl 5, 10, 15, etc.)", false))
+        config.registerOption("multipleOf10", Toggle("Send only multiples of 10", "Only send multiples of 10 (Lvl 10, 20, 30, etc.)", false))
+        config.registerOption("useRomanNumerals", Toggle("Use Roman Numerals", "Use Roman Numerals instead of Arabic Numerals in the message", false))
+    }
+
+    val regex = "SKILL LEVEL UP (\\w+) (\\w+)➜(\\w+)".toRegex()
+    @SubscribeEvent
+    fun onChatMessage(event: ClientChatReceivedEvent) {
+        val message = event.trueUnformattedMessage
+
+        if (!regex.containsMatchIn(message)) {
+            return
+        }
+
+        val (skill, oldLevel, newLevel) = regex.find(message)?.destructured ?: return
```

Comment:
no need to perform the regex check twice
If it doesnt match, destructured should be null making it return